### DISCUSSION TITLE
Can't update nvim-cmp

### DISCUSSION THREAD

**Discussion by ghost** [lumen-oss/rocks.nvim#483] [Q&A] [answered]
Don't really know if this is a rocks issue, a nvim-cmp issue, or due to my workaround. I am the same guy from #342, still using the same workaround to get rocks to work on chimera linux. When trying to update plugins, all plugins update successfully, except nvim-cmp, which fails. unfortunately the logs aren't particularly helpful.
```
1   [START][2024-07-24 18:57:40] rocks.nvim logging initiated
  1 ERROR | 2024-07-24 18:58:40 | ....local/share/nvim/rocks/share/lua/5.1/rocks/luarocks.lua:65 | luarocks CLI FAILED
  2 ERROR | 2024-07-24 18:58:40 | ....local/share/nvim/rocks/share/lua/5.1/rocks/luarocks.lua:66 | Warning: Failed searching manifest: Failed downloading https://nvim-          neorocks.github.io/rocks-binaries/dev/manifest-5.1 - failed downloading https://nvim-neorocks.github.io/rocks-binaries/dev/manifest-5.1
  3 
  4 Error: No results matching query were found for Lua 5.1.
  5 To check if it is available for other Lua versions, use --check-lua-versions.
  6 
  7 ERROR | 2024-07-24 18:58:40 | ...re/nvim/rocks/share/lua/5.1/rocks/operations/helpers.lua:93 | Failed to install nvim-cmp
  8 ERROR | 2024-07-24 18:58:40 | ...are/nvim/rocks/share/lua/5.1/rocks/operations/update.lua:77 | UPDATE ERROR: Failed to update nvim-cmp.
  9 ERROR | 2024-07-24 18:59:09 | ...are/nvim/rocks/share/lua/5.1/rocks/operations/update.lua:177 | Update completed with errors! Run ':Rocks log' for details.
```

**Comment by mrcjkb** [comment]
Hey :wave: 

we had to remove nvim-cmp from the luarocks.org root manifest, because the last release tag (`v0.0.1`) is extremely outdated (almost 2 years old).
Unfortunately, all our attempts to get a more recent release published have been ignored so far.

See

- https://github.com/hrsh7th/nvim-cmp/pull/1923
- https://github.com/hrsh7th/nvim-cmp/issues/1880

Sorry for the inconvenience :disappointed: 

It should be possible to install the `dev` rock with `:Rocks install nvim-cmp dev`.

**Reply by ghost** [reply]
Thanks for the response, when I run `:Rocks install nvim-cmp dev` it fails with `Could not find nvim-cmp dev Installation failed`
Logs are still seemingly irrelevant
```
[START][2024-07-24 23:07:55] rocks.nvim logging initiated
ERROR | 2024-07-24 23:08:02 | ....local/share/nvim/rocks/share/lua/5.1/rocks/luarocks.lua:65 | luarocks CLI FAILED
ERROR | 2024-07-24 23:08:02 | ....local/share/nvim/rocks/share/lua/5.1/rocks/luarocks.lua:66 | Warning: Failed searching manifest: Failed downloading https://nvim-      neorocks.github.io/rocks-binaries/dev/manifest-5.1 - failed downloading https://nvim-neorocks.github.io/rocks-binaries/dev/manifest-5.1

Error: No results matching query were found for Lua 5.1.
To check if it is available for other Lua versions, use --check-lua-versions.

ERROR | 2024-07-24 23:08:02 | ...re/nvim/rocks/share/lua/5.1/rocks/operations/helpers.lua:93 | Failed to install nvim-cmp
~
```

**Reply by mrcjkb** [reply, marked as answer]
It looks like the upload didn't publish to the root manifest.
Maybe because there's another [(non-root) package](https://luarocks.org/modules/teto/nvim-cmp) :thinking: 

EDIT: Nevermind, I just managed to add it to the root manifest manually.
It should work now.

**Reply by teto** [reply]
https://luarocks.org/search?q=nvim-cmp doesnt list mine on the root manifest. but the dev version remains. Does it mean Rocks picked my version over neorocks' ? I can totally remove mine but it would be nice to have away to avoid that and for instance select the neorocks one if there is a doubt

**Reply by mrcjkb** [reply]
@teto see my edit :)

**Reply by ghost** [reply]
Yup, fixed now, thanks
